packages/axelar-wasm-std/src/chain_name.rs

```diff
@@ -0,0 +1,461 @@
+use std::fmt;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use const_str::contains;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{StdError, StdResult};
+use cw_storage_plus::{Key, KeyDeserialize, Prefixer, PrimaryKey};
+use error_stack::report;
+use valuable::Valuable;
+
+use crate::{FnExt, FIELD_DELIMITER};
+
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("chain name is invalid")]
+    InvalidChainName,
```

Comment:
InvalidChainName is still an error defined in router-api, mostly because of the constructor for CrossChainId. It's very difficult to get around that. Would it make more sense to just move the entire `primitives.rs` into this package? (separate PR)

It's kind of odd that the ChainName types specifically are done here while all the other core types live in the router-api